crates/telemetry/src/config.rs

```diff
@@ -0,0 +1,72 @@
+///
+/// Largely consists of OpenTelemetry variables defined
+/// [here](https://opentelemetry.io/docs/specs/otel/configuration/sdk-environment-variables/#general-sdk-configuration)
+/// but also has some Spin specific configuration.
+#[derive(Debug)]
+pub struct Config {
+    /// Disables OpenTelemetry for all signals (traces and metrics).
+    pub otel_sdk_disabled: bool,
+
+    /// Determines the verbosity of the OpenTelemetry tracing layer.
+    ///
+    /// This is a Spin specific value in the sense that it is not defined by the OpenTelemetry
+    /// standard. This is what allows us to have different levels of verbosity in the fmt layer and
+    /// the otel layer. The fmt layer respects `RUST_LOG` and the otel layer respects
+    /// `SPIN_OTEL_TRACING_LEVEL`.
+    ///
+    /// Note: Currently SPIN_OTEL_TRACING_LEVEL defaults to info and makes no distinction between
+    /// spans from Spin crates or external dependencies. In the future we may desire to provide a
+    /// more curated experience of what spans show by default. This curated experience could be
+    /// created by annotating every Spin span with a special attribute that can be filtered on. Then
+    /// if the user opts to set SPIN_OTEL_TRACING_LEVEL this curated filter would be turned off and
+    /// we would allow spans from any library at the level defined by the user.
+    pub otel_tracing_level: Level,
+
+    /// Spin version.
+    pub spin_version: String,
+}
+
+impl Config {
+    /// Initializes a new [Config] from the environment.
+    pub fn from_env(version: String) -> anyhow::Result<Self> {
+        let inner = EnvDerivableConfig::init_from_env()?;
+        Ok(Self {
+            otel_sdk_disabled: inner.otel_sdk_disabled,
+            otel_tracing_level: inner.otel_tracing_level,
+            spin_version: version,
+        })
+    }
+
+    /// Returns a boolean indicating if the OTEL layer should be enabled.
+    ///
+    /// It is considered enabled if any of the following environment variables are set:
+    /// - `OTEL_EXPORTER_OTLP_ENDPOINT`
+    /// - `OTEL_EXPORTER_OTLP_TRACES_ENDPOINT`
+    /// - `OTEL_EXPORTER_OTLP_METRICS_ENDPOINT`
+    ///
+    /// Note that this is overridden if OTEL_SDK_DISABLED is set to true.
+    pub fn otel_enabled() -> bool {
+        const ENABLING_VARS: &[&str] = &[
+            "OTEL_EXPORTER_OTLP_ENDPOINT",
+            "OTEL_EXPORTER_OTLP_TRACES_ENDPOINT",
+            "OTEL_EXPORTER_OTLP_METRICS_ENDPOINT",
+        ];
+        ENABLING_VARS
+            .iter()
+            .any(|key| std::env::var_os(key).is_some_and(|val| !val.is_empty()))
+    }
+}
+
+/// EnvDerivableConfig represents the elements of Config that are derived from environment variables.
+#[derive(Envconfig, Debug)]
+struct EnvDerivableConfig {
+    #[envconfig(from = "OTEL_SDK_DISABLED", default = "false")]
+    pub otel_sdk_disabled: bool,
+
+    #[envconfig(from = "SPIN_OTEL_TRACING_LEVEL", default = "info")]
+    pub otel_tracing_level: Level,
+}
```

Comment:
If this is all we're using `envconfig` for I'd suggest dropping it and using `std::env::var` directly in `from_env`.